content/posts/2019-05-02-apollo-client-in-practice.md

```diff
@@ -7,14 +7,25 @@ tags:
     - webdevelopment
 ---
 
-Some time ago I joined a team that is working on a search application. Implemented in React and using [Apollo](https://www.apollographql.com/) for 
-GraphQL calls, I was surprised to find it noticeably slow when a lot of search results were retrieved. ...expand...
+Some time ago I joined a team that is working on a search application. The application takes search terms and displays 
+the results in a table with potentially dozens of columns and hundreds 
+ of rows, even before pagination. It is implemented in React and uses [Apollo](https://www.apollographql.com/) for 
+GraphQL calls. I was surprised to find it noticeably slow when a lot of search results were retrieved. React is 
+well-known for leveraging virtual DOM to optimize performance and GraphQL should even be able to add caching to further
+optimize performance on the side of network requests. 
 
 Looking into the performance tab of Chrome dev tools lead me to believe the performance problems were caused by 
 computations in the bottom components (e.g. formatting in cells). Because there are so many and they are 
 re-rendered quite often, this approach is quite intensive on resources.
 
-Besides that, the application had obvious state synchronization problems. When moving between views it was not  maintaining the same state of selected rows. Even though Redux is used to store application state and communicate it between components, it is not used consistently. There are still plenty of React class components that store some parts of the state locally. 
+Besides that, the application had obvious state synchronization problems. When moving between views it was not 
+maintaining the same state of selected rows. Even though Redux is used to store application state and communicate it 
+between components, it is not used consistently. There are still plenty of React class components that store some parts of the state locally. 
```

Comment:
`it is not used consistently` -> `it was not used consistently`?

Reply:
I have the tendency to "forget" what tense I am writing in. I fixed it in some other spots.

Reply:
A very common problem.